tests/python/pants_test/backend/python/tasks/test_ctypes_integration.py

```diff
@@ -122,21 +122,13 @@ def test_ctypes_third_party_integration(self):
     self.assert_success(pants_run)
     self.assertIn('Test worked!\n', pants_run.stdout_data)
 
-    # Test cached run.
```

Comment:
To test a cached run, you could use something like: https://github.com/pantsbuild/pants/blob/master/tests/python/pants_test/backend/jvm/tasks/jvm_compile/test_dep_exports_integration.py#L35

ie, run pants twice in the same temporary workdir.

Reply:
Ack, will add.

Reply:
Is that test checking for a cached run by checking whether the output contains the target spec (the last line of that method)? Where is that message coming from? How does that technique compare to the use of `self.mock_buildroot()` as used in [another method also named `test_invalidation()`](https://github.com/pantsbuild/pants/blob/90c145a03fad24ec8024cddbce3b2a8b1c75f819/tests/python/pants_test/backend/python/tasks/test_python_distribution_integration.py#L64)?

Reply:
@stuhood after reviewing the conan code further, I see that this test is unnecessary because really the only thing that needs to be tested is whether the invalidation check does what it is supposed to.